### PR TITLE
Implement Path merging

### PR DESCRIPTION
**Pull request by hello-josh**
Also fixed a bug with an invalid mock assertion.

### CODE REVIEW

**Comment by sloria** [comment]
Tests are failing due to incompatibility with webargs>=0.11.0. I will fix this shortly.

**Comment by sloria** [comment]
@Trii OK, updating with the latest `dev` branch should fix the build.

**Comment by hello-josh** [comment]
oops, I didn't realize I still my local webargs fork in my virtualenv. I was wondering why it failed on travis.

File: tests/apispec/test_core.py [hunk 1]
```diff
@@ -105,6 +105,32 @@ def test_add_path(self, spec):
...
+                    responses=route_spec['responses'],
+                    produces=route_spec['produces'],
+                    operationId='updatePet',
+                    summary='Updates an existing Pet',
+                    description='Use this method to make changes to Pet `petId`',
+                    tags=route_spec['tags']
+                )
+            )
+        )
+
```
Comment (sloria) [inline tests/apispec/test_core.py:133]: Is there supposed to be an assertion here?

Comment (hello-josh) [inline tests/apispec/test_core.py:133]: derp, the next talk started and I forgot to add it. I'll do it now.

Comment (hello-josh) [inline tests/apispec/test_core.py:133]: Done!

**Comment by sloria** [comment]
:+1:

**Pull request merged**